apps/emqx_management/test/emqx_mgmt_api_clients_SUITE.erl

```diff
@@ -171,6 +200,257 @@ t_clients(_) ->
+                list_request(APIPort, "node=" ++ atom_to_list(Node)),
+                #{node => Node}
+            )
+        ),
+        ?assertMatch(
+            {ok, {{_, 200, _}, _, #{<<"connected">> := IsConnected}}},
+            lookup_request(ClientId, APIPort)
+        ),
+        ok
+    end,
+
+    ?assertMatch({ok, {{_, 200, _}, _, #{<<"data">> := []}}}, list_request(APIPort)),
+
+    ?check_trace(
+        begin
+            %% Scenario 1
+            %% 1) Client connects and is listed as connected.
+            %% 2) Client disconnects and is listed as disconnected.
+            %% 3) Client reconnects and is listed as connected.
+            %% 4) Client disconnects.
+            %% 5) Session is GC'ed, client is removed from list.
+            ?tp(notice, "scenario 1", #{}),
+            ClientId1 = <<"c1">>,
+            C1 = Connect(#{port => Port1, clientid => ClientId1}),
+            AssertSingleClient(N1, ClientId1, connected),
+            ok = emqtt:disconnect(C1),
+            AssertSingleClient(N1, ClientId1, disconnected),
+            C2 = Connect(#{port => Port1, clientid => ClientId1}),
+            AssertSingleClient(N1, ClientId1, connected),
+            ok = emqtt:stop(C2),
+            %% simulate GC
+            ?tp(notice, "gc", #{}),
+            ok = erpc:call(N1, emqx_persistent_session_ds, destroy_session, [ClientId1]),
+            ?retry(
+                100,
+                20,
+                ?assertMatch(
+                    {ok, {{_, 200, _}, _, #{<<"data">> := []}}},
+                    list_request(APIPort)
+                )
+            ),
+
+            %% Scenario 2
+            %% 1) Client connects and is listed as connected.
+            %% 2) Client connects to the same node and takes over, listed only once.
+            ?tp(notice, "scenario 2", #{}),
+            ClientId2 = <<"c2">>,
+            _C3 = Connect(#{port => Port1, clientid => ClientId2}),
+            AssertSingleClient(N1, ClientId2, connected),
+            process_flag(trap_exit, true),
+            C4 = Connect(#{port => Port1, clientid => ClientId2}),
+            AssertSingleClient(N1, ClientId2, connected),
+            ok = emqtt:stop(C4),
+            ok = erpc:call(N1, emqx_persistent_session_ds, destroy_session, [ClientId2]),
+            ?retry(
+                100,
+                20,
+                ?assertMatch(
+                    {ok, {{_, 200, _}, _, #{<<"data">> := []}}},
+                    list_request(APIPort)
+                )
+            ),
+
+            %% Scenario 3
+            %% 1) Client connects and is listed as connected.
+            %% 2) Client connects to *another node* and takes over, listed only once.
+            ?tp(notice, "scenario 3", #{}),
+            ClientId3 = <<"c3">>,
+            _C5 = Connect(#{port => Port1, clientid => ClientId3}),
+            AssertSingleClient(N1, ClientId3, connected),
+            process_flag(trap_exit, true),
```

Comment:
Nit. I vaguely remember CT runner having problems with stuck and/or failing testcases that set `trap_exit` to `true`.